cocos/2d/CCNode.h

```diff
@@ -1881,9 +1906,16 @@ class CC_DLL Node : public Ref
     mutable bool _additionalTransformDirty; ///< transform dirty ?
     bool _transformUpdated;         ///< Whether or not the Transform object was updated since the last frame
 
-    int _localZOrder;               ///< Local order (relative to its siblings) used to sort the node
+#if CC_64BITS
+    std::int64_t _localZOrder;               ///< Local order (relative to its siblings) used to sort the node
+#else
+    int _localZOrder; /// < Local order (relative to its siblings) used to sort the node
+    unsigned int _orderOfArrival; /// < used to preserve sequence while sorting children with the same localZOrder
+#endif
```

Comment:
how about declare 3 variables no matter on all architecture like this:
```c++
std::int64_t _localZOrderAndArrival;
int _localZRrder;
unsigned int _orderOfArrival;
```

These 3 variables keep synchronous all the time but differently used is used for optimization in `nodeComparisonLess `.